src/commonMain/kotlin/com/algolia/search/model/search/Query.kt

```diff
@@ -382,172 +383,172 @@ public data class Query(
      * Engine default: [ResponseFields.All]
      * [Documentation][https://www.algolia.com/doc/api-reference/api-parameters/responseFields/?language=kotlin]
      */
-    @SerialName(KeyResponseFields) var responseFields: List<ResponseFields>? = null,
+    @SerialName(KeyResponseFields) override var responseFields: List<ResponseFields>? = null,
 
     /**
      * Maximum number of facet hits to return during a search for facet values.
      * Engine default: 10
      * [Documentation][https://www.algolia.com/doc/api-reference/api-parameters/maxFacetHits/?language=kotlin]
      */
-    @SerialName(KeyMaxFacetHits) var maxFacetHits: Int? = null,
+    @SerialName(KeyMaxFacetHits) override var maxFacetHits: Int? = null,
 
     /**
      * Whether to include or exclude a query from the processing-time percentile computation.
      * Engine default: true
      * [Documentation][https://www.algolia.com/doc/api-reference/api-parameters/percentileComputation/?language=kotlin]
      */
-    @SerialName(KeyPercentileComputation) var percentileComputation: Boolean? = null,
+    @SerialName(KeyPercentileComputation) override var percentileComputation: Boolean? = null,
 
     /**
      *  Overrides the query parameter and performs a more generic search that can be used to find "similar" results.
      *  Engine default: ""
      *  [Documentation][https://www.algolia.com/doc/api-reference/api-parameters/similarQuery/?language=kotlin]
      */
-    @SerialName(KeySimilarQuery) var similarQuery: String? = null,
+    @SerialName(KeySimilarQuery) override var similarQuery: String? = null,
 
     /**
      * Whether this query should be taken into consideration by currently active ABTests.
      * Engine default: true
      * [Documentation][https://www.algolia.com/doc/api-reference/api-parameters/enableABTest/?language=kotlin]
      */
-    @SerialName(KeyEnableABTest) var enableABTest: Boolean? = null,
+    @SerialName(KeyEnableABTest) override var enableABTest: Boolean? = null,
 
     /**
      * Enriches the API’s response with meta-information as to how the query was processed.
      * It is possible to enable several [ExplainModule] independently.
      * Engine default: null
      * [Documentation][https://www.algolia.com/doc/api-reference/api-parameters/decompoundedAttributes/?language=kotlin]
      */
-    @SerialName(KeyExplain) var explainModules: List<ExplainModule>? = null,
+    @SerialName(KeyExplain) override var explainModules: List<ExplainModule>? = null,
 
     /**
      * List of supported languages with their associated language ISO code.
      * Provide an easy way to implement voice and natural languages best practices such as ignorePlurals,
      * removeStopWords, removeWordsIfNoResults, analyticsTags and ruleContexts.
      */
-    @SerialName(KeyNaturalLanguages) var naturalLanguages: List<Language>? = null
-)
+    @SerialName(KeyNaturalLanguages) override var naturalLanguages: List<Language>? = null,
+) : SearchParameters
```

Comment:
I would create a separate structure, sort of `SearchParametersImpl` and make `Query: SearchParameters by SearchParametersImpl` to put `AnswersQuery` and `Query` (meaning `SearchQuery`) as well as other structures implementing `SearchParameters` on the same level of abstraction and to make them independent from each other.
Right now `Query` is a "pure" implementation of `SearchParameters`, but it may happen that we will need to add something specific to `Query` that is logically not a part of other structures depending on `Query`.  So the creation of a separate implementation of `SearchParameters` that will always stay pure is reasonable. 
WDYT?

Reply:
Yup, I've tried it, but unfortunately, this approach will break the backward compatibility of `Query` (because its a `data class`) :/
We can still add specific fields to `Query` without adding them to `SearchParameters` with the current approach.

Reply:
I see. Let's keep it as is so 👍